tdesign/src/themes/withMUITheme.tsx

```diff
@@ -8,11 +11,15 @@ import UserTheme from "./UserTheme";
 // If multiple instances of Emotion + caching becomes a problem again, see:
 // https://github.com/emotion-js/emotion/issues/2210#issuecomment-758577800
 
-const withMUITheme = (Page) => {
-  const WithMUITheme = (props) => {
+const withMUITheme = <PageContext extends {}, IP extends {}, Props extends {}>(
+  Page: NextComponentType<PageContext, IP, Props>
+) => {
+  const WithMUITheme: NextComponentType<PageContext, IP, Props> = (props) => {
     return (
       <UserTheme>
-        <Page {...props} />
+        {/* NOTE: Props are not modified in any way - we can safely ignore
+          type-checking error here */}
+        <Page {...(props as any)} />
```

Comment:
Is the reason to `as any` here to avoid expensive type checking like in other places?

Reply:
Not this time. Here it was some type mismatch with a very obscure error message that I didn't want to get into, knowing that it is safe :smile: 

![image](https://user-images.githubusercontent.com/889383/141488729-f4e2edf5-674b-4aa6-99e6-9ddff6840833.png)